### PR TITLE
Added lombok for better logs, cleaner variables, better exceptions

### PR DESCRIPTION
**Pull request by RajatArora08**

### CODE REVIEW

File: src/main/java/appointment/watcher/Watcher.java [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 import com.twilio.type.PhoneNumber;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class Watcher {
 
+    private static final String TWILIO_ACCOUNT_SID = "<twilio account sid>";
+    private static final String TWILIO_AUTH_TOKEN = "<twilio auth token>";
+    private static final String FROM_PHONE_NUMBER = "<from phone number>"; // Select from Twilio
+    private static final String TO_PHONE_NUMBER = "<to phone number>";
```
Comment (jeevjyot) [inline src/main/java/appointment/watcher/Watcher.java:16]: May be we can move this into config file?

File: src/main/java/appointment/watcher/Watcher.java [hunk 2]
```diff
@@ -18,35 +27,35 @@ public static void main(String[] args) throws Exception {
                 //TODO: send text only when appointment is available
                 available = watcherService.checkIfAppointmentAvailable();
                 sendText("Appointment available = " + available);
-            } catch (Exception e) {
-                System.out.println(e.getCause());
-                sendText("Something went wrong=" + e.getMessage());
+            } catch (CustomException e) {
```
Comment (jeevjyot) [inline src/main/java/appointment/watcher/Watcher.java]: We can catch very specific exception? to convey what is CustomException is about?

File: src/main/java/appointment/watcher/Watcher.java [hunk 3]
```diff
@@ -18,35 +27,35 @@ public static void main(String[] args) throws Exception {
...
+                    log.info(e.getMessage());
+                } else {
+                    log.error("Error occurred: {}", e.getMessage());
+                    sendText("Something went wrong=" + e.getMessage());
+                }
             }
-            Thread.sleep(10_000); //feel free to change the frequency
-        }
 
+            Thread.sleep(10000); //feel free to change the frequency
```
Comment (jeevjyot) [inline src/main/java/appointment/watcher/Watcher.java]: Underscores in numbers help in readability!

File: src/main/java/appointment/watcher/clients/AppointmentClient.java [hunk 4]
```diff
@@ -29,12 +34,16 @@ public String getAppointmentResponse() throws Exception {
...
 
-        if (response.statusCode() != 200) {
-            System.out.println("non successful response");
-            System.out.println("StatusCode =" + response.statusCode());
-            System.out.println("ResponseBody=" + response.body());
-            throw new Exception("Failed to get the response " + response.statusCode());
+        if (response.statusCode() == 302 &&
+                response.headers().map().get("location").get(0).contains("Clinic+does+not+have+any+appointment+slots+available.")) {
+
+            throw new CustomException(ExceptionType.SKIPPABLE, "No appointments.");
```
Comment (jeevjyot) [inline src/main/java/appointment/watcher/clients/AppointmentClient.java]: Lets throw AppointmentNotFoundException (a custom exception)?

**Review by jeevjyot — CHANGES_REQUESTED** [review]

**Comment by jeevjyot** [comment]
Thanks for this PR and making changes @RajatArora08 !!

File: src/main/java/appointment/watcher/Watcher.java (hunk 1 above)
Comment (RajatArora08) [reply src/main/java/appointment/watcher/Watcher.java:16]: Not using Spring here. Any easy way to read a config file in core java? I don't feel like writing a parser for a .config file lol

Comment (jeevjyot) [reply src/main/java/appointment/watcher/Watcher.java:16]: Ah, I meant we can just move to constants file for now! Yes, we probably can use properties file but that's okay for now!

**Comment by RajatArora08** [comment]
@jeevjyot Added new changes. Converted all TODO to git issues. Also, message is now sent only when appointment is available

**Review by jeevjyot — APPROVED** [review]

**Comment by jeevjyot** [comment]
Merging now! Thanks for making the changes.

**Pull request merged**